### ISSUE TITLE
Using kind images with SHA digest in e2e

### ISSUE THREAD

**Issue by daman1807** [kubernetes-sigs/kpng#510]
### What kind of issue is this?
Currently in E2E we pass the k8s version: v1.25.3/ v1.27.1 as an argument and `./hack/test_e2e.sh` spawns a kind cluster with `docker.io/kindest/node:v1.25.3` and `docker.io/kindest/node:v1.27.1` respectively. Instead of passing tags, we should be passing tags with their SHA digests from the latest kind releases - https://github.com/kubernetes-sigs/kind/releases.

In short: The[ value for the image tag ](https://github.com/kubernetes-sigs/kpng/blob/e48c8d7f74212607ab151195397617dc8d146e2c/hack/test_e2e.sh#LL275C39-L275C39
) should include the sha as well.

**Comment by clearnicki** [comment]
I can help with this. 

/assign
